testing/git_cached_repository/git_cached_repository.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+
+source ./testing/e2e/init.sh # Must be sourced 1st
+source ./testing/e2e/bazel.sh
+source ./testing/e2e/rules.sh
+source ./testing/e2e/unittest.sh
+source ./testing/e2e/logging.sh
+source ./testing/e2e/utils.sh
+source ./testing/e2e/repos.sh
+source ./testing/e2e/assert.sh
+source ./testing/e2e/testkit.sh
+
+# Must be invoked from the root of the repo.
+ROOT=$PWD
+
+function before_test() {
+  set_up
+  TEST_name=$1
+  TEST_passed="true"
+  create_clean_workspace
+  copy_git_cached_repo_rules_to_workspace
+  set_git_cached_repo_path
+  log_test_name ${TEST_name}
+  # Always cd into the workspace root directory
+  cd ${WORKSPACE_DIR}
+}
+
+function after_test() {
+  tear_down
+  delete_git_cached_directory ${TEST_GIT_CACHE_DIR}
+}
+
+function test_cached_dir_created_as_expected() {
+  before_test "test_cached_dir_created_as_expected"
+
+  # Given I declare a git cache rule with custom cache directory
+  create_git_cached_rule_for_small_repo "014459e6361b66a7b210758d4bf93f3a46ca5e88" ${TEST_GIT_CACHE_DIR}
```

Comment:
what is this `014459e6361b66a7b210758d4bf93f3a46ca5e88`? random?

Reply:
It is the commit-hash used for the `commit` attribute of `git_cached_repository` rule.
I've created a var named `commit_hash` though it seems a bit verbose, specifically when inlining bash function calls